pkg/api/steve/supportconfigs/generator.go

```diff
@@ -30,12 +30,18 @@ const (
 	// Pay-As-You-Go (PAYG) offerings. However, they will be in different namespaces to avoid
 	// clashing.
 	cspAdapterConfigmap = "csp-config"
-	tarContentType      = "application/x-tar"
-	logPrefix           = "support-config-generator"
+	// Metering archive for CSP marketplace Pay-As-You-Go (PAYG
+	// offerings. Used for auditing purposes.
+	cspMeteringArchiveConfigmap = "metering-archive"
+	tarContentType              = "application/x-tar"
+	logPrefix                   = "support-config-generator"
 )
 
 var errNotFound = errors.New("not implemented")
 
+// indicating whether user is authorized to access the metering archive
+var authorizedForMeteringArchive = false
```

Comment:
By including this as a global variable you are persisting the state across different user requests. This could create a race condition where one user's request allows another user's request. Do not use a global variable for this.

Reply:
Good catch! Will do.